src/Angular.js

```diff
@@ -859,7 +859,12 @@ function shallowCopy(src, dst) {
 
   return dst || src;
 }
-
+function equalsCacheContains(cache, o1, o2) {
+  for (var i = 0, ii = cache.length; i < ii; i += 2) {
+    if (cache[i] === o1 && cache[i + 1] === o2) return true;
```

Comment:
This might be faster (not tested):

```js
var index = 0, length = cache.length;
while (index < length) {
  index = cache.indexOf(o1, index);
  if (index == -1) break;
  if (cache[index + 1] === o2) return true;
}
return false;
```